core/src/test/scala/integration/kafka/api/ProxyBasedFederationTest.scala

```diff
@@ -0,0 +1,74 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional information regarding copyright
+ * ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy of the
+ * License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied. See the License for the specific language governing permissions and limitations under the
+ * License.
+ */
+
+package kafka.api
+
+import scala.collection.JavaConverters._
+
+import org.junit.Assert._
+import org.junit.Test
+
+import kafka.server.KafkaServer
+import kafka.server.QuotaType
+
+/**
+ * Currently a simple proof of concept of a multi-cluster integration test, but ultimately intended
+ * as a feasibility test for proxy-based federation:  specifically, can a vanilla client (whether
+ * consumer, producer, or admin) talk to a federated set of two or more physical clusters and
+ * successfully execute all parts of its API?
+ */
+class ProxyBasedFederationTest extends MultiClusterAbstractConsumerTest {
+  override def numClusters: Int = 2  // need one ZK instance for each Kafka cluster [FIXME: can we "chroot" instead?]
+  override def brokerCountPerCluster: Int = 3  // three _per Kafka cluster_, i.e., six total
+
+  @Test
+  def testBasicMultiClusterSetup(): Unit = {
+    val numRecords = 1000
+    val producer = createProducer()
+    sendRecords(producer, numRecords, tp1)
+
+    val consumer = createConsumer()
+    consumer.assign(List(tp1).asJava)
+    consumer.seek(tp1, 0)
+    consumeAndVerifyRecords(consumer = consumer, numRecords = numRecords, startingOffset = 0)
+
+
+    // this stuff is just ripped off from PlaintextConsumerTest testQuotaMetricsNotCreatedIfNoQuotasConfigured()
+    def assertNoMetric(broker: KafkaServer, name: String, quotaType: QuotaType, clientId: String): Unit = {
```

Comment:
I feel we should focus on the core functionality of producing, consuming, admin operations for now, and leave these metrics testing out (at least for now). What do you think?

Reply:
Oh, sure.  These are just copy-pasta from the PlaintextConsumerTest so there were some explicit assertions, not anything specifically relevant to federation.